lib/HeroBonus.h

```diff
@@ -522,6 +523,16 @@ struct DLL_LINKAGE Bonus : public std::enable_shared_from_this<Bonus>
 		return (high << 16) + low;
 	}
 
+	STRONG_INLINE static ui32 getHighFromSid32(ui32 sid)
+	{
+		return sid >> 16;

```

Comment:
```suggestion
		return sid & 0xFFFF0000;
```

Reply:
But that's different effect? Bit shift != bit mask?

Reply:
@IvanSavenko agree.

Reply:
@Nordsoft91 we should not to use & here, because `0xdeadbeef & 0xffff0000 = 0xdead0000`, but `0xdeadbeef >> 16 = 0xdead`, and I need second behaviour.